### PR TITLE
align imageNames with component names

### PR DESCRIPTION
**Pull request by ozdanborne**
## Description

There is a bit of inconsistency around our concept of 'components'. This PR makes things more consistent by tying them together. Here's how the source of truth works now:

- A component 'name' as specified in versions.yml can be treated as the 'id' of that component.
- The docker image for a given component can now be looked up via imageNames, since `imageNames[componentName] = docker image name`
- The name of an image could potentially change between releases. As such, it is now a per-directory setting instead of a global, site-wide setting.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit d26420ac97a120b6f6dc6f2a7e74597ec6d590b4

https://deploy-preview-2806--calico.netlify.com

File: _includes/component_url [hunk 1]
```diff
@@ -1,16 +1,20 @@
 {%- if release.title != "master" -%}
 {%- case include.component -%}
+{%- comment -%}
+note that all 'or's in the below case statement are there to maintain backwards compatability 
+with how image names used to be referenced in config.yml
+{%- endcomment -%}
 
-{%- when 'calico/node' -%}
+{%- when 'node' or 'calico/node' -%}
```
Comment (bcreane) [inline _includes/component_url]: Are you checking both the old and new key values in order to maintain upgrade path?

Comment (ozdanborne) [reply _includes/component_url]: yup. the comment I added above was supposed to explain that :smile:

Comment (bcreane) [reply _includes/component_url]: oops, yeah.

**Pull request merged**